antismash/modules/rrefinder/rrefinder.py

```diff
@@ -0,0 +1,199 @@
+        if record.id != self.record_id:
+            raise ValueError("Record to store in and record analysed don't match")
+
+        for feature in self.features:
+            record.add_feature(feature)
+
+    def to_json(self) -> Dict[str, Any]:
+        """ Constructs a JSON representation of this instance """
+
+        return {
+            "schema_version": self.schema_version,
+            "bitscore_cutoff": self.bitscore_cutoff,
+            "hits_per_protocluster": self.hits_per_protocluster,
+            "hit_info": self.hit_info,
+            "min_length": self.min_length,
+            "record_id": self.record_id,
+        }
+
+    @staticmethod
+    def from_json(json: Dict[str, Any], record: Record, min_length: int, bitscore_cutoff: float
+                  ) -> Optional["RREFinderResults"]: # type: ignore  # pylint: disable=arguments-differ
+        """ Regenerates the RREFinderResults from json.
+            If less strict settings are given (e.g. a lower cutoff
+            or a shorter minimum length), the results are discarded.
+        """
+
+        # check that the previous data version is the same as current, if not, discard the results
+        if json["schema_version"] != RREFinderResults.schema_version:
+            return None
+
+        if record.id != json.get("record_id"):
+            logging.warning("RREFinder results are for different record, discarding previous results")
+            return None
+
+        prev_min_length = json.get("min_length")
+        prev_bitscore_cutoff = json.get("bitscore_cutoff")
+
+        if prev_min_length is None or prev_bitscore_cutoff is None:
+            raise ValueError('Invalid RREfinderResults json dictionary')
+        assert isinstance(prev_min_length, int) and isinstance(prev_bitscore_cutoff, float)
+
+        # Check if the cutoff options set in this run are the same or stricter than in the previous run
+        if bitscore_cutoff < prev_bitscore_cutoff:
+            logging.debug("RREFinderResults bitscore cutoff has changed, discarding previous results")
+            return None
+        elif min_length < prev_min_length:
+            logging.debug("RREFinderResults minimum length has changed, discarding previous results")
+            return None
+
+        # Refilter the hits (in case the cutoff is now more stringent)
+        filtered_hit_info, filtered_hits_per_protocluster = filter_hits(json['hit_info'],
+                                                                        json['hits_per_protocluster'],
+                                                                        min_length, bitscore_cutoff)
+        return RREFinderResults(record.id, bitscore_cutoff, min_length,
+                                filtered_hits_per_protocluster, filtered_hit_info)
+
+
+def is_ripp(product: str) -> bool:
+    """ Determines which BGC products are RiPPs
+    """
+    ripp_products = ['bacteriocin', 'cyanobactin', 'lanthipeptide',
```

Comment:
Every time this function is called, it needs to create this list. It would be better as a constant outside the function to avoid that. It would also be better as a set instead of a list, as the time complexity would be constant instead of linear.